src/calculation/calculation.js

```
@@ -0,0 +1,40 @@
+import { validateData } from "./validator.js"
+
+const getTopLikedVideos = (data) =>
+    validateData(data)
+        ? data
+              .sort((videoA, videoB) => videoB.likeCount - videoA.likeCount)
+              .slice(0, 5)
+        : []
+
+const getNewestVideos = (data) =>
+    validateData(data)
+        ? data
+              .sort(
+                  (videoA, videoB) =>
+                      new Date(videoB.publishedAt) -
+                      new Date(videoA.publishedAt)
+              )
+              .slice(0, 5)
+        : []
+
+const getTotalLikes = (data) =>
+    validateData(data)
+        ? data.reduce((total, video) => total + parseInt(video.likeCount), 0)
+        : (() => { throw new Error("Invalid input: Expected an array of videos with its metadata"); })();
```

This is not great!
Why create an anonymous function and call it immediately?
You can simply throw the error.
Why parsing it?
Why not return an empty array?